codecs/text/src/test/java/com/datastax/oss/dsbulk/codecs/text/string/StringToIntegerCodecTest.java

```diff
@@ -27,22 +27,61 @@
 
 class StringToIntegerCodecTest {
 
-  private StringToIntegerCodec codec;
+  private StringToIntegerCodec codec1;
+  private StringToIntegerCodec codec2;
 
   @BeforeEach
-  void setUp() {
+  void setUpCodec1() {
+    ConvertingCodecFactory codecFactory = new ConvertingCodecFactory(new TextConversionContext());
+    codec1 =
+        (StringToIntegerCodec)
+            codecFactory.<String, Integer>createConvertingCodec(
+                DataTypes.INT, GenericType.STRING, true);
+  }
+
+  @BeforeEach
+  void setUpCodec2() {
     ConversionContext context =
-        new TextConversionContext().setNullStrings("NULL").setFormatNumbers(true);
+        new TextConversionContext().setNullStrings("NULL", "NADA").setFormatNumbers(true);
     ConvertingCodecFactory codecFactory = new ConvertingCodecFactory(context);
-    codec =
+    codec2 =
         (StringToIntegerCodec)
             codecFactory.<String, Integer>createConvertingCodec(
                 DataTypes.INT, GenericType.STRING, true);
   }
 
   @Test
   void should_convert_from_valid_external() {
-    assertThat(codec)
+    assertThat(codec1)
+        .convertsFromExternal("0")
+        .toInternal(0)
+        .convertsFromExternal("2147483647")
+        .toInternal(Integer.MAX_VALUE)
+        .convertsFromExternal("-2147483648")
+        .toInternal(Integer.MIN_VALUE)
+        .convertsFromExternal("2,147,483,647")
+        .toInternal(Integer.MAX_VALUE)
+        .convertsFromExternal("-2,147,483,648")
+        .toInternal(Integer.MIN_VALUE)
+        .convertsFromExternal("2,147,483,647")
+        .toInternal(Integer.MAX_VALUE)
+        .convertsFromExternal("-2,147,483,648")
+        .toInternal(Integer.MIN_VALUE)
+        .convertsFromExternal("1970-01-01T00:00:00Z")
+        .toInternal(0)
+        .convertsFromExternal("TRUE")
+        .toInternal(1)
+        .convertsFromExternal("FALSE")
+        .toInternal(0)
+        .convertsFromExternal("")
```

Comment:
If the input is an empty string, it is converted to null since this is a non-textual target CQL type `int`.